src/fixed.c

```diff
@@ -12,39 +12,59 @@
 
 twin_fixed_t twin_fixed_sqrt(twin_fixed_t a)
 {
-    twin_fixed_t max = a, min = 0;
-
-    while (max > min) {
-        twin_fixed_t mid = (max + min) >> 1;
-        if (mid >= 181 * TWIN_FIXED_ONE) {
-            max = mid - 1;
-            continue;
-        }
-        twin_fixed_t sqr = twin_fixed_mul(mid, mid);
-        if (sqr == a)
-            return mid;
-        if (sqr < a)
-            min = mid + 1;
-        else
-            max = mid - 1;
+    if (a <= 0)
+        return 0;
+    if (a < TWIN_FIXED_ONE + (1 << (8 - 1)) &&
```

Comment:
You can introduce a macro for range checks. See https://github.com/sysprog21/semu/blob/master/common.h#L20-L27